agent/engine/docker_task_engine.go

```diff
@@ -1752,26 +1781,72 @@ func (engine *DockerTaskEngine) provisionContainerResources(task *apitask.Task,
 		})
 		return dockerapi.DockerContainerMetadata{
 			DockerID: cniConfig.ContainerID,
-			Error: ContainerNetworkingError{errors.Wrapf(err,
-				"container resource provisioning: failed to setup network namespace")},
+			Error: ContainerNetworkingError{fmt.Errorf(
+				"container resource provisioning: failed to setup network namespace: %+v", err)},
 		}
 	}
 
 	return dockerapi.MetadataFromContainer(containerInspectOutput)
 }
 
-// checkTearDownPauseContainerAwsvpc idempotently tears down the pause container network when the pause container's known
+func (engine *DockerTaskEngine) provisionContainerResourcesBridgeMode(task *apitask.Task, container *apicontainer.Container) dockerapi.DockerContainerMetadata {
+	if !task.IsServiceConnectEnabled() || container.Type != apicontainer.ContainerCNIPause {
+		return dockerapi.DockerContainerMetadata{
+			Error: ContainerNetworkingError{fromError: fmt.Errorf(
+				"container resource provisioning bridge mode: cannot setup netns - only valid for SC-enabled task pause container"),
+			},
+		}
+	}
+
+	containerInspectOutput, err := engine.inspectContainer(task, container)
+	if err != nil || containerInspectOutput == nil {
+		return dockerapi.DockerContainerMetadata{
+			Error: ContainerNetworkingError{fromError: fmt.Errorf(
+				"container resource provisioning bridge mode: cannot setup netns - error inspecting container %s: %+v", container.Name, err),
+			},
+		}
+	}
+
+	cniConfig, err := engine.buildCNIConfigFromTaskContainerBridgeMode(task, containerInspectOutput, container.Name)
+	if err != nil {
+		return dockerapi.DockerContainerMetadata{
+			Error: ContainerNetworkingError{fromError: fmt.Errorf(
+				"container resource provisioning bridge mode: unable to build cni configuration for container %s: %+v", container.Name, err),
+			},
+		}
+	}
+
+	// Invoke the libcni to config the network namespace for the container
+	_, err = engine.cniClient.SetupNS(engine.ctx, cniConfig, cniSetupTimeout)
+	log.Infof("container resource provisioning bridge mode for %s: just invoked CNI plugin", container.Name)
+
+	if err != nil {
+		logger.Error("Unable to configure pause container namespace", logger.Fields{
+			field.TaskID: task.GetID(),
+			field.Error:  err,
+		})
+		return dockerapi.DockerContainerMetadata{
+			DockerID: cniConfig.ContainerID,
+			Error:    ContainerNetworkingError{fmt.Errorf("container resource provisioning: failed to setup network namespace: %+v", err)},
+		}
+	}
+
+	log.Infof("Successfully configured pause netns for %s", container.Name)
+	return dockerapi.MetadataFromContainer(containerInspectOutput)
+}
+
+// checkTearDownPauseContainer idempotently tears down the pause container network when the pause container's known
 //or desired status is stopped.
-func (engine *DockerTaskEngine) checkTearDownPauseContainerAwsvpc(task *apitask.Task) {
-	if !task.IsNetworkModeAWSVPC() {
+func (engine *DockerTaskEngine) checkTearDownPauseContainer(task *apitask.Task) {
```

Comment:
nit: to tie to the pause container naming thread -- here you're not differentiating between the sc pause container and the base pause container.  If this should only apply to the ServiceConnect specific pause container I would call this `checkTearDownServiceConnectPauseContainer()`... 

however my preference (as I noted above) is just reusing the `NetworkPauseContainerName` and keeping it generic.

Reply:
> you're not differentiating between the sc pause container and the base pause container

That's actually the intention. The branching for networking mode comes in at a later point inside `cleanupPauseContainerNetwork` method, similar to how we branch in `provisionContainerResources`